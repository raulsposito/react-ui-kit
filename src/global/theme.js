// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
