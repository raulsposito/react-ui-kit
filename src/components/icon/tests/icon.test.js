import React from 'react'
import { render, screen } from '@testing-library/react'

import { colors, spacing } from '@global/theme'

import Icon from '../index'

describe('<Icon />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Icon name='face' />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Disabled', () => {
    it('is not disabled', () => {
      render(<Icon name='face' />)
      const i = screen.getByText('face')

      expect(i).toHaveStyle({ color: `${colors.contrast.regular}` })
    })

    it('is not disabled and have color', () => {
      render(<Icon name='face' color='variant' />)
      const i = screen.getByText('face')

      expect(i).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('is disabled', () => {
      render(<Icon name='face' disabled />)
      const i = screen.getByText('face')

      expect(i).toHaveAttribute('disabled')
      expect(i).toHaveStyle({ color: `${colors.contrast.disabled}` })
    })

    it('is disabled and have color', () => {
      render(<Icon name='face' color='variant' disabled />)
      const i = screen.getByText('face')

      expect(i).toHaveAttribute('disabled')
      expect(i).toHaveStyle({ color: `${colors.variant.disabled}` })
    })
  })

  describe('Sizes', () => {
    it('is xsmall', () => {
      render(<Icon size='xsmall' name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.oneAndAHalf}` })
      expect(i).toHaveStyle({ width: `${spacing.oneAndAHalf}` })
      expect(i).toHaveStyle({ height: `${spacing.oneAndAHalf}` })
    })

    it('is small', () => {
      render(<Icon size='small' name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.two}` })
      expect(i).toHaveStyle({ width: `${spacing.two}` })
      expect(i).toHaveStyle({ height: `${spacing.two}` })
    })

    it('is medium', () => {
      render(<Icon name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.twoAndAHalf}` })
      expect(i).toHaveStyle({ width: `${spacing.twoAndAHalf}` })
      expect(i).toHaveStyle({ height: `${spacing.twoAndAHalf}` })
    })

    it('is large', () => {
      render(<Icon size='large' name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.three}` })
      expect(i).toHaveStyle({ width: `${spacing.three}` })
      expect(i).toHaveStyle({ height: `${spacing.three}` })
    })

    it('is xlarge', () => {
      render(<Icon size='xlarge' name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.threeAndAHalf}` })
      expect(i).toHaveStyle({ width: `${spacing.threeAndAHalf}` })
      expect(i).toHaveStyle({ height: `${spacing.threeAndAHalf}` })
    })

    it('is xxlarge', () => {
      render(<Icon size='xxlarge' name='face' />)
      const i = screen.getByText('face')
      expect(i).toHaveStyle({ 'font-size': `${spacing.four}` })
      expect(i).toHaveStyle({ width: `${spacing.four}` })
      expect(i).toHaveStyle({ height: `${spacing.four}` })
    })
  })

  describe('Colors', () => {
    it('is contrast', () => {
      render(<Icon name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.contrast.regular}` })
    })

    it('is primary', () => {
      render(<Icon color='primary' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.primary.regular}` })
    })

    it('is variant', () => {
      render(<Icon color='variant' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('is secondary', () => {
      render(<Icon color='secondary' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.secondary.regular}` })
    })

    it('is success', () => {
      render(<Icon color='success' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.success.regular}` })
    })

    it('is errror', () => {
      render(<Icon color='error' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.error.regular}` })
    })

    it('is warning', () => {
      render(<Icon color='warning' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.warning.regular}` })
    })

    it('is system', () => {
      render(<Icon color='system' name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ color: `${colors.system.regular}` })
    })
  })

  describe('Shape', () => {
    it('does NOT has a circle', () => {
      render(<Icon name='face' />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ 'border-radius': '' })
      expect(icon).toHaveStyle({ 'box-shadow': '' })
    })

    it('has a circle', () => {
      render(<Icon name='face' circle />)
      const icon = screen.getByText('face')
      expect(icon).toHaveStyle({ 'border-radius': '50%' })
      expect(icon).toHaveStyle({ 'box-shadow': '0 0 0 3px #3C3C3C' })
    })
  })
})
