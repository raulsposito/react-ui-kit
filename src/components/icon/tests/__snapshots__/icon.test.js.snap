// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-bdfBQB iUuvkC material-icons"
  >
    face
  </span>
</DocumentFragment>
`;
